pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -547,55 +536,29 @@ pub mod pallet {
 			domain: Domain,
 			message: T::Message,
 		) -> (DispatchResult, Weight) {
-			let mut weight = T::DbWeight::get().reads(1);
+			let read_weight = T::DbWeight::get().reads(1);
 
-			let router = match DomainRouters::<T>::get(domain) {
-				Some(r) => r,
-				None => return (Err(Error::<T>::RouterNotFound.into()), weight),
+			let Some(router) = DomainRouters::<T>::get(domain) else {
+				return (Err(Error::<T>::RouterNotFound.into()), read_weight);
 			};
 
-			match router.send(sender.clone(), message.serialize()) {
-				Ok(dispatch_info) => Self::update_total_post_dispatch_info_weight(
-					&mut weight,
-					dispatch_info.actual_weight,
-				),
-				Err(e) => {
-					Self::update_total_post_dispatch_info_weight(
-						&mut weight,
-						e.post_info.actual_weight,
-					);
-
-					return (Err(e.error), weight);
-				}
-			}
+			let serialized = message.serialize();
+			let serialized_len = serialized.len() as u64;
 
-			(Ok(()), weight)
-		}
-
-		/// Updates the provided `PostDispatchInfo` with the weight required to
-		/// process an outbound message.
-		pub(crate) fn update_total_post_dispatch_info_weight(
-			weight: &mut Weight,
-			router_call_weight: Option<Weight>,
-		) {
-			let router_call_weight =
-				Self::get_outbound_message_processing_weight(router_call_weight);
-
-			*weight = weight.saturating_add(router_call_weight);
-		}
+			// TODO: do we really need to return the weights in send() if later we use the
+			// defensive ones?
+			let (result, router_weight) = match router.send(sender, serialized) {
+				Ok(dispatch_info) => (Ok(()), dispatch_info.actual_weight),
+				Err(e) => (Err(e.error), e.post_info.actual_weight),
+			};
 
-		/// Calculates the weight used by a router when processing an outbound
-		/// message.
-		fn get_outbound_message_processing_weight(router_call_weight: Option<Weight>) -> Weight {
-			let pov_weight: u64 = (Domain::max_encoded_len()
-				+ T::AccountId::max_encoded_len()
-				+ T::Message::max_encoded_len())
-			.try_into()
-			.expect("can calculate outbound message POV weight");
-
-			router_call_weight
-				.unwrap_or(Weight::from_parts(LP_DEFENSIVE_WEIGHT_REF_TIME, 0))
-				.saturating_add(Weight::from_parts(0, pov_weight))
+			(
+				result,
+				router_weight
+					.unwrap_or(Weight::from_parts(LP_DEFENSIVE_WEIGHT_REF_TIME, 0))
+					.saturating_add(read_weight)
+					.saturating_add(Weight::from_parts(0, serialized_len)),
```

Comment:
NOTE we need here the serialized len instead of `max_decoded_len()` because the size used internally by the router is the size of the vector. The vector doesn't know about `Message` type